### PR TITLE
Change quick access naming

### LINKED ISSUE #37: Fix Quick access & method/class parameters
# Description
Our client has given us feedback on the relevance of the parameters in our quick access window. Some of the parameters that we have included in the previous sprint are actually rarely (if at all) modified (like e.g `local search budget`). This issue will focus on incorporating their feedback and correcting the problems.
 
The fixes should ensure that the code is consistent throughout the entire code base, i.e. if we are removing a certain parameter from the quick access then we should make sure that it is removed correctly in all the places where it is no longer relevant/used. 

# Definition of Done
* [x]  The name of the section (the title) has changed from "Frequently used parameters" to  "**Quick Access Parameters**"
* [x]  The parameters have been placed in the **correct order**, according to their priorities (e.g `search budget` is the first);
* [x]  Clear **sections** for the groups of the parameters have been created, according to the client (*search budget*, *timeouts*, *population*)
* [x]  `search budget` and `stopping condition` have been placed in own category (since they are related). 
* [x]  It has been made clear that `stopping condition` is referring to the _budget type_ of `search budget` (rename or extend a tooltip) 
* [x]  `search budget` has a dynamic unit depending on the selected type
* [x]  *Parameters* tab has been placed to the last place in the tool window
* [x]  The code works and compiles

### PR DESCRIPTION
**Pull request by jzelenjak**
# Description of changes made
* The name of the section (the title) has been changed from "Frequently used parameters" to  "**Quick Access Parameters**"
* The parameters have been placed in the **correct order**, according to their priorities (e.g `search budget` is the first);
* Clear **sections** for the groups of the parameters have been created (*Search Budget*, *Timeouts*, *Genetic Algorithm*), 
* `stopping condition` has been renamed to _search budget type_ in the label that the user can see (not in the code)
* `search budget` and `population` have a dynamic unit depending on the selected type
* *Parameters* tab has been placed to the last place in the tool window
*  Tooltips have been added to the actual UI elements, not only the labels for them

# Why is merge request needed
This merge request is needed since it addresses the feedback given by the client.

Closes #37 

# Other notes

# What is missing?
* Not something that is crucial to do, but something that might be useful to do: make the default values of the parameters available via a properties file to improve maintainability

- [x] I have checked that I am merging into correct branch

### CODE REVIEW

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt [hunk 1]
```diff
@@ -1,49 +1,67 @@
...
+        stoppingCondition.addActionListener {
+            stoppingConditionToolTip.text = default("60 ${stoppingCondition.item.units()}")
+        }
+        stoppingConditionToolTip.text = default("60 ${stoppingCondition.item.units()}")
+
+        populationLimitToolTip.border = JBUI.Borders.emptyLeft(10)
+        populationLimit.addActionListener {
+            populationLimitToolTip.text = default("60 ${populationLimit.item.toString().toLowerCase()}")
+        }
+        populationLimitToolTip.text = default("60 ${populationLimit.item.toString().toLowerCase()}")
```
Comment (martinmladenov) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:64]: There seems to be a little bit of code duplication here. Could that be fixed somehow?

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt [hunk 2]
```diff
@@ -1,49 +1,67 @@
...
 
+    // The tooltip labels
+    private var stoppingConditionToolTip = JBLabel("Default: 60 seconds", UIUtil.ComponentStyle.SMALL, UIUtil.FontColor.BRIGHTER)
+    private var populationLimitToolTip = JBLabel("Default: 60 seconds", UIUtil.ComponentStyle.SMALL, UIUtil.FontColor.BRIGHTER)
+
+    // Template strings for "default" tooltips
     private val defaultStr: String = "Default: %s"
 
+
     init {
```
Comment (martinmladenov) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:46]: Just a small comment: it would be nice to add come comments to the constructor to make the different sections easier to distinguish, or even better, move the code for initialising different sections to separate private methods.

**Review by martinmladenov — COMMENTED** [review]
Now it looks very good and intuitive, nice work! I'm sure the clients will love it.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt [hunk 3]
```diff
@@ -1,49 +1,67 @@
...
+    // UI elements for EvoSuite parameters
     private var stoppingCondition: ComboBox<StoppingCondition> = ComboBox<StoppingCondition>(StoppingCondition.values())
+    private var searchBudget: JSpinner = JSpinner(SpinnerNumberModel(60, 0, 10000, 1))
     private var initializationTimeout: JSpinner = JSpinner(SpinnerNumberModel(120, 0, 10000, 1))
     private var minimisationTimeout: JSpinner = JSpinner(SpinnerNumberModel(60, 0, 10000, 1))
     private var assertionTimeout: JSpinner = JSpinner(SpinnerNumberModel(60, 0, 10000, 1))
     private var junitCheckTimeout: JSpinner = JSpinner(SpinnerNumberModel(60, 0, 10000, 1))
-    private var population: JSpinner = JSpinner(SpinnerNumberModel(50, 0, 10000, 1))
     private var populationLimit: ComboBox<PopulationLimit> = ComboBox(PopulationLimit.values())
+    private var population: JSpinner = JSpinner(SpinnerNumberModel(50, 0, 10000, 1))
```
Comment (bolkhod) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:28]: Have we not agreed to move population to advanced settings (as per meeting mattermost post of client)?

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindowState.kt [hunk 4]
```diff
@@ -6,45 +6,35 @@ package nl.tudelft.ewi.se.ciselab.testgenie.toolwindow
...
-        const val localSearchBudgetValue: Int = 5
         const val initializationTimeout: Int = 120
         const val minimizationTimeout: Int = 60
         const val assertionTimeout: Int = 60
         const val junitCheckTimeout: Int = 60
-        const val population: Int = 50
-        val localSearchBudgetType: LocalSearchBudgetType = LocalSearchBudgetType.TIME
-        val stoppingCondition: StoppingCondition = StoppingCondition.MAXTIME
         val populationLimit: PopulationLimit = PopulationLimit.INDIVIDUALS
+        const val population: Int = 50
```
Comment (bolkhod) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindowState.kt:26]: Out of interesting, why did you move state values around? Does it matter? (I mean `population`)

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt [hunk 5]
```diff
@@ -56,66 +74,32 @@ class TestGenieToolWindow {
         .setFormLeftIndent(30)
         .addVerticalGap(25)
         .addComponent(panelTitle)
+        .addComponent(JXTitledSeparator("Search budget"), 35)
+        .addLabeledComponent(customLabel("Search budget type",
+            "What condition should be checked to end the search."), stoppingCondition, 25, false)
+        .addTooltip(default(StoppingCondition.MAXTIME.toString()))
         .addLabeledComponent(customLabel("Search budget", "Maximum search duration."), searchBudget, 25, false)
-        .addTooltip(default("60 seconds"))
```
Comment (bolkhod) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:60]: General remark: This section will need to be double checked when merging this branch's changes with #1

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt [hunk 6]
```diff
@@ -226,16 +206,30 @@ class TestGenieToolWindow {
...
+        stoppingCondition.toolTipText = "What condition should be checked to end the search."
+        searchBudget.toolTipText = "Maximum search duration."
+        initializationTimeout.toolTipText = "Seconds allowed for initializing the search."
+        minimisationTimeout.toolTipText = "Seconds allowed for minimization at the end."
+        assertionTimeout.toolTipText = "Seconds allowed for assertion generation at the end."
+        junitCheckTimeout.toolTipText = "Seconds allowed for checking the generated JUnit files <p/>(e.g., compilation and stability)."
+        populationLimit.toolTipText = "What to use as limit for the population size."
+        population.toolTipText = "Population size of genetic algorithm."
+
+        resetButton.toolTipText = "Reset all parameters to their default values"
```
Comment (bolkhod) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:232]: Were these in your general constructor before? You've had tooltips prior to this, no?

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindowState.kt (hunk 4 above)
Comment (SergeyDatskiv) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindowState.kt:26]: We moved the state values around because the interpretation of the value depends on the the type that was selected, therefore it makes sense to first select the type and then change the value. 

E.g.
PopulationLimit = Individuals 
Then the population value is 50 Individuals
OR
PopulationLimit = Statements
Then the population value is 50 Statements

I think it is more intuitive if the category is before the value. 

If this is what you asked.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt (hunk 3 above)
Comment (SergeyDatskiv) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:28]: As we discussed in the morning. The clients mattermost is outdated and in your notes "population" is in the quick access settings. Although, this was a confusing point during the meeting, so I wouldn't be surprised if we need to change it later.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt (hunk 6 above)
Comment (SergeyDatskiv) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:232]: We did have tooltips before, however now we refactored it to make the tooltips clearer to see.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt (hunk 1 above)
Comment (SergeyDatskiv) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:64]: This comment was addressed by Jegor and you in the code. I think that everything should be fine now.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt (hunk 2 above)
Comment (SergeyDatskiv) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:46]: This comment was addressed by Jegor and you in the code. I think that everything should be fine now.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt [hunk 7]
```diff
@@ -1,49 +1,67 @@
...
+import com.intellij.util.ui.UIUtil
+import org.jdesktop.swingx.JXTitledSeparator
 import java.awt.*
+import java.awt.event.ActionEvent
 import javax.swing.*
 
 /**
  * This class stores the UI of the TestGenie tool window.
  */
 class TestGenieToolWindow {
```
Comment (martinmladenov) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:18]: Maybe also change the names of the classes since this doesn't involve the entire tool window but only the parameters tab.

Comment (SergeyDatskiv) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:18]: This comment was addressed by Jegor and you in the code. I think that everything should be fine now.

**Review by martinmladenov — APPROVED** [review]
Everything looks good to me now. :)﻿

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt (hunk 5 above)
Comment (jzelenjak) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/toolwindow/TestGenieToolWindow.kt:60]: I have even triple-checked that

**Review by bolkhod — APPROVED** [review]
Overall a clean improvement on the client's wishes

**Review by martinmladenov — APPROVED** [review]
